### PR TITLE
CB-11032 adding instance filter for too small instances which has les…

### PR DESCRIPTION
**Pull request by doktoric**
…s then 4 cpu and 16 gb memory

See detailed description in the commit message.

### CODE REVIEW

File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/filter/MinimalHardwareFilter.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import com.sequenceiq.cloudbreak.cloud.model.VmType;
+
+@Service
+public class MinimalHardwareFilter {
+
+    private static final int MIN_CPU = 4;
+
+    private static final int MIN_MEMORY = 16;
+
+    public boolean enoughAsMinimumHardware(VmType vmType) {
```
Comment (lajosrodek) [inline cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/filter/MinimalHardwareFilter.java:14]: Nit: Perhaps rename it to `suitableAsMinimumHardware()`.

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpPlatformResources.java [hunk 2]
```diff
@@ -444,6 +450,9 @@ public CloudVmTypes getCloudVmTypes(CloudCredential cloudCredential, Region regi
                     }
                 }
 
+                types = types.stream()
+                        .filter(e -> minimalHardwareFilter.suitableAsMinimumHardware(e))
+                        .collect(Collectors.toSet());
```
Comment (lajosrodek) [inline cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpPlatformResources.java:455]: This change will also apply to DL clusters (as called from `virtualMachines()`). Is that what you want?

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpPlatformResources.java [hunk 3]
```diff
@@ -396,11 +400,13 @@ public CloudVmTypes virtualMachinesForDistroX(CloudCredential cloudCredential, R
             for (Map.Entry<String, Set<VmType>> stringSetEntry : cloudVmResponses.entrySet()) {
                 returnVmResponses.put(stringSetEntry.getKey(), stringSetEntry.getValue().stream()
                         .filter(enabledDistroxInstanceTypeFilter)
+                        .filter(e -> minimalHardwareFilter.suitableAsMinimumHardware(e))
```
Comment (lajosrodek) [inline cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpPlatformResources.java:403]: Probably it is a bug, but the `virtualMachines()` call in line 396 (via `getCloudVmTypes()`) already takes care of instance filtering (for DL as well). So either the change in `getCloudVmTypes()` is not needed, or this change and the one in line 409 are not needed.

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpPlatformResources.java [hunk 4]
```diff
@@ -396,11 +400,13 @@ public CloudVmTypes virtualMachinesForDistroX(CloudCredential cloudCredential, R
...
                         .collect(Collectors.toSet()));
             }
         } else {
             for (Map.Entry<String, Set<VmType>> stringSetEntry : cloudVmResponses.entrySet()) {
                 returnVmResponses.put(stringSetEntry.getKey(), stringSetEntry.getValue().stream()
+                        .filter(e -> minimalHardwareFilter.suitableAsMinimumHardware(e))
```
Comment (lajosrodek) [inline cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpPlatformResources.java:409]: See comment in line 403.

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpPlatformResources.java [hunk 5]
```diff
@@ -396,11 +400,13 @@ public CloudVmTypes virtualMachinesForDistroX(CloudCredential cloudCredential, R
...
                         .collect(Collectors.toSet()));
             }
         }
```
Comment (lajosrodek) [inline cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpPlatformResources.java:412]: This comment is for lines 416-417 but GH does not let me add it there. So, `@NotNull` shall be removed (the agreement was that we were not going to use JetBrains annotations), and `getCloudVmTypes()` shall be made `private` (or package private if directly called by UT).

File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/VmTypeMeta.java [hunk 6]
```diff
@@ -71,6 +71,16 @@ public void setSt1Config(VolumeParameterConfig st1Config) {
         return properties;
     }
 
+    public Integer getCPU() {
+        Object cpuAsObject = properties.get(CPU);
+        return cpuAsObject != null ? Integer.valueOf(cpuAsObject.toString()) : null;
+    }
+
+    public Float getMemory() {
```
Comment (lajosrodek) [inline cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/VmTypeMeta.java:79]: Perhaps add suffix `InGB`.

File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/filter/MinimalHardwareFilter.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.cloud.model.VmType;
+
+@Service
+public class MinimalHardwareFilter {
+
+    private static final int MIN_CPU = 4;
+
+    private static final int MIN_MEMORY = 16;
```
Comment (lajosrodek) [inline cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/filter/MinimalHardwareFilter.java:12]: Perhaps add suffix `_IN_GB`.

File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/filter/MinimalHardwareFilter.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+    public boolean suitableAsMinimumHardware(VmType vmType) {
+        boolean enough = false;
```
Comment (lajosrodek) [inline cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/filter/MinimalHardwareFilter.java:15]: Please rename to `suitable`, `response` or something more appropriate. 😃

**Review by lajosrodek — CHANGES_REQUESTED** [review]
Mostly minor improvements, but the GCP part may have a potential bug.

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpPlatformResources.java (hunk 3 above)
Comment (doktoric) [reply cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpPlatformResources.java:403]: removing virtual machine filter thanks

File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/filter/MinimalHardwareFilter.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.cloud.model.VmType;
+
+@Service
+public class MinimalHardwareFilter {
+
+    private static final int MIN_CPU = 4;
+
+    private static final int MIN_MEMORY_ING_GB = 16;
```
Comment (lajosrodek) [inline cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/filter/MinimalHardwareFilter.java:12]: Typo: ING -> IN.

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpPlatformResources.java [hunk 10]
```diff
@@ -396,18 +399,19 @@ public CloudVmTypes virtualMachinesForDistroX(CloudCredential cloudCredential, R
...
         return new CloudVmTypes(returnVmResponses, cloudVmTypes.getDefaultCloudVmResponses());
     }
 
-    @NotNull
     public CloudVmTypes getCloudVmTypes(CloudCredential cloudCredential, Region region, Map<String, String> filters) {
```
Comment (lajosrodek) [inline cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpPlatformResources.java:415]: Nit: To be made `private`.

**Review by lajosrodek — APPROVED** [review]
LGTM, up to you to fix or ignore these bits. Approving nevertheless.

**Pull request merged**